### ISSUE TITLE
Build charms making invalid legal claims

### ISSUE THREAD

**Issue by stub42** [juju/charm-tools#241]
## Checklist

 - [X] Confirmed this is an issue with charm-tools, not charmstore-client
 - [X] Provide versions of tools used
 - [X] Described the feature or ways to replicate the issue

## What version am I running?

I ran the following command: `charm version` and got the following ouput:

```
charm 2.1.1-0ubuntu1
charm-tools 2.1.2
```

I am using: Ubunut

## Issue/Feature

Build needs to handle copyright files correctly.

- If a charm does not include a copyright file, when composed one from one of its layers is used instead. charm proof doesn't complain, and invalid legal assertions are made

- When composing a charm, copyright statements from layers are thrown away, possibly in violation of licenses like BSD

Instead, charm build needs to collate and keep the copyright files in the composed charm. Possibly by just renaming them copyright.$layer, or sticking them in a licenses directory.

### I expect/expected the following

No bugs and world peace.

### What I got

A face palm from legal.

**Comment by chuckbutler** [comment]
Yeah, we had some preliminary discussions about this early on, and I'm pretty sure we punted on it at the time. There was talk of license linting and an even more complex implementation that scanned for complimentary license options... I'm not sure what happened to that thread.

**Comment by stub42** [comment]
I'd start simple and iterate, since perfect is impossible in this case.

1. Keep copyright statements from layers
1. Move to machine readable format
1. Report licenses at build time, so the charmer can make an informed decision
1. Magic system to detect and block conflicts at build time based on the charmers current legal jurisdiction.
1. A pony.

This issue is calling out for the first step.
